### PR TITLE
Add Scale harmonization feature

### PR DESCRIPTION
**Pull request by mystery-house**
This pull request adds two new methods to the `Scale` class, `harmonize()` and `harmonize_dict()`. 

Each one calculates a list of diatonic chords per note in the scale, plus dominant 7th chords.

Dominant 7th chords are included by default, but can be omitted with an argument.

`harmonize()` returns a list of lists, where the outer index corresponds to the index of each note in the scale. `harmonize_dict()` returns a dict of lists, where the dict keys correspond to note names. For examples, see the JSON files in the `tests` directory.

Also included are two example scripts which can be used to update the test JSON files if new scale or chord recipes are added in the future.

### CODE REVIEW

File: tests/harmonized_scales_list.json
Comment (gciruelos) [inline tests/harmonized_scales_list.json:1]: do we really need 8k lines of tests? can you please reduce this to at most 200 lines testing some representative examples?

**Review by gciruelos — CHANGES_REQUESTED** [review]
thank you for the contribution! It looks good, please resolve the comments and I will merge it.

File: tests/harmonized_scales_list.json
Comment (gciruelos) [reply tests/harmonized_scales_list.json:1]: looking at them more closely, i think something around 40-50 lines per json file should be enough

Comment (mystery-house) [reply tests/harmonized_scales_list.json:1]: Yep. I erred on the side of completeness, but I'll scale it back.

File: musthe/musthe.py [hunk 1]
```diff
@@ -534,3 +534,64 @@ def __str__(self):
...
+        Args:
+            include_dom7 (bool, optional): If ``True``, dominant 7th chords will be included, if applicable. Defaults to ``True``.
+
+        Returns:
+            List[Union[List[Chord], None]]: A ``list`` containing either matching :py:class:`Chord` objects for each scale degree, or ``None`` if no chords could be found.
+
+        See Also:
+            - :py:meth:`Scale.harmonize_dict`
+        """
+        chords = [None]*len(self.notes)
```
[the remark this answers is not in the document]
Comment (mystery-house) [reply musthe/musthe.py]: Done.

File: tests/harmonized_scales_list.json
Comment (mystery-house) [reply tests/harmonized_scales_list.json:1]: I removed the large JSON files in favor of tests for `harmonize()` and `harmonize_dict()` with and without the `include_dom7` flag; each one tests a different scale, and it tests the expected `None` result when no chords can be matched, so it should be pretty representative.

**Comment by gciruelos** [comment]
Merged, thank you for your contribution.

**Pull request merged**